frontend/src/tests/applicant-tests.js

```diff
@@ -126,7 +126,67 @@ export function applicantTests(api) {
         expect(resp.payload).toContainObject(newApplicant);
     });
 
-    it.todo("delete an applicant with no associated assignments");
+    it("delete an applicant with no associated assignments", async () => {
+        // upsert and delete an applicant without a session
+        const newApplicant1 = {
+            first_name: "June",
+            last_name: "Kim",
+            utorid: "kim",
+        };
+        const resp = await apiPOST(`/admin/applicants`, newApplicant1);
+        expect(resp).toHaveStatus("success");
+        Object.assign(newApplicant1, resp.payload);
+
+        // try deleting the applicant
+        const resp2 = await apiPOST(`/admin/applicants/delete`, {
+            id: newApplicant1.id,
+        });
+        expect(resp2).toHaveStatus("success");
 
-    it.todo("fail to delete an applicant with an associated assignment");
+        // verify the applicant is deleted
+        const resp3 = await apiGET(`/admin/applicants`);
+        expect(resp3).toHaveStatus("success");
+        expect(resp3.payload.map((x) => x.id)).not.toContain(newApplicant1.id);
```

Comment:
I'd feel safer if you double checked that `newApplicant1.id != null` before you did this check.
Since this would be an error in the test setup, you can just `throw new Error(...)` if it's null.

Reply:
I wonder if checking ·newApplicant1.id != null` is really necessary here because if it verifies that the response status is "success" above it, right?